cibuildwheel/linux.py

```diff
@@ -85,40 +89,48 @@ def build(project_dir, output_dir, test_command, test_requires, before_build, bu
                 fi
                 delocated_wheel=(/tmp/delocated_wheel/*.whl)
 
-                # Install the wheel we just built
-                "$PYBIN/pip" install "$delocated_wheel"
+                # install tox to run the tests in an isolated environment
+                $PYBIN/pip install tox --ignore-installed
 
-                # Install any requirements to run the tests
-                if [ ! -z "{test_requires}" ]; then
-                    "$PYBIN/pip" install {test_requires}
-                fi
+                test_dir=`mktemp -d`
 
-                # Run the tests from a different directory
-                if [ ! -z {test_command} ]; then
-                    pushd $HOME
-                    PATH="$PYBIN:$PATH" sh -c {test_command}
-                    popd
-                fi
+                cat > $test_dir/tox.ini << EOL
+            [tox]
+            envlist = test
+            skipsdist = true
+
+            [testenv]
+            whitelist_externals = *
+            commands =
+                pip install $delocated_wheel
+                {test_requires_cmd}
+                {test_cmd_prepared}
+            EOL
```

Comment:
I'm not familiar with tox, so bear with me - what is tox doing for us here? I was expecting to see some virtualenv commands, but I guess tox is handling that on our behalf using these options. Could it perhaps be clearer by doing running the virtualenv commands directly? 

(In my head it's just a `virtualenv $test_dir/env`, then a subshell that calls `source $test_dir/env/bin/activate` before the package-specific stuff.... that feels clearer to me, and requires less knowledge)

Reply:
Apologies, I now see your note in the PR. I can't see the previous attempts using direct virtualenv calls in this PR, I'd be curious what issues you ran into.

Reply:
@joerick - I still have a backup of the virtualenv version locally so can push it to a branch for you to look at shortly. Essentially the issue is that sometimes when the virtualenv was set up, ``python -m pip`` would still point to the system install even though ``python`` pointed to the virtualenv version, and I couldn't find a way around it.

Reply:
@astrofrog please do - I might be able to help you on that!

Reply:
@joerick - I just opened https://github.com/joerick/cibuildwheel/pull/164 which has my original attempt with virtualenv alone.